File: scripts/split_logo.py
"""
Split the ASON logo image into separate components:
- logo-symbol.png: The brush-style A symbol with the golden dot
- logo-text.png: The "ASON" brand name
- logo-slogan-en.png: "A Sense Of Now"
- logo-slogan-cn.png: "从现在做起，永远不迟。"
- logo-full.png: Original full image

Usage:
    python scripts/split_logo.py <input_image_path> [--output-dir <output_dir>]
"""

import argparse
import os
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("Error: Pillow is required. Install it with: pip install Pillow")
    exit(1)


def split_logo(input_path: str, output_dir: str) -> None:
    img = Image.open(input_path)
    w, h = img.size
    print(f"Input image size: {w} x {h}")

    # Define crop regions (top, bottom) as percentages of height
    # Based on visual analysis of the ASON logo layout
    regions = {
        "logo-symbol.png": (0.10, 0.52),      # Mountain/A symbol + golden dot
        "logo-text.png": (0.52, 0.65),          # "ASON" brand name
        "logo-slogan-en.png": (0.65, 0.73),     # "A Sense Of Now"
        "logo-slogan-cn.png": (0.73, 0.85),     # "从现在做起，永远不迟。"
    }

    os.makedirs(output_dir, exist_ok=True)

    for filename, (top_pct, bottom_pct) in regions.items():
        top = int(h * top_pct)
        bottom = int(h * bottom_pct)
        # Add small padding
        padding = int(h * 0.01)
        top = max(0, top - padding)
        bottom = min(h, bottom + padding)

        cropped = img.crop((0, top, w, bottom))
        output_path = os.path.join(output_dir, filename)
        cropped.save(output_path, "PNG")
        print(f"Saved: {output_path}  ({cropped.size[0]}x{cropped.size[1]})")

    # Also save a transparent version of just the symbol
    symbol_top = int(h * 0.10)
    symbol_bottom = int(h * 0.52)
    symbol_img = img.crop((0, symbol_top, w, symbol_bottom))

    # Attempt to remove white background
    symbol_transparent = symbol_img.convert("RGBA")
    data = symbol_transparent.getdata()
    new_data = []
    threshold = 240
    for item in data:
        if item[0] > threshold and item[1] > threshold and item[2] > threshold:
            new_data.append((255, 255, 255, 0))
        else:
            new_data.append(item)
    symbol_transparent.putdata(new_data)
    transparent_path = os.path.join(output_dir, "logo-symbol-transparent.png")
    symbol_transparent.save(transparent_path, "PNG")
    print(f"Saved: {transparent_path}  ({symbol_transparent.size[0]}x{symbol_transparent.size[1]})")

    # Save full copy
    full_path = os.path.join(output_dir, "logo-full.png")
    img.save(full_path, "PNG")
    print(f"Saved: {full_path}  (full copy)")

    print("\nDone! All components saved to:", output_dir)


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Split ASON logo into components")
    parser.add_argument("input", help="Path to the input logo image")
    parser.add_argument(
        "--output-dir",
        default="public/static/images/logo",
        help="Output directory (default: public/static/images/logo)",
    )
    args = parser.parse_args()
    split_logo(args.input, args.output_dir)

File: scripts/split_logo.mjs
import sharp from 'sharp'
import path from 'path'
import fs from 'fs'

const input = process.argv[2]
const outputDir = process.argv[3] || 'public/static/images/logo'

if (!input) {
  console.error('Usage: node scripts/split_logo.mjs <input_image> [output_dir]')
  process.exit(1)
}

const img = sharp(input)
const meta = await img.metadata()
const w = meta.width
const h = meta.height
console.log(`Input: ${w}x${h}`)

fs.mkdirSync(outputDir, { recursive: true })

// Crop regions: [top%, bottom%]
const regions = [
  { name: 'logo-symbol.png', top: 0.1, bottom: 0.52 },
  { name: 'logo-text.png', top: 0.52, bottom: 0.65 },
  { name: 'logo-slogan-en.png', top: 0.65, bottom: 0.73 },
  { name: 'logo-slogan-cn.png', top: 0.73, bottom: 0.85 },
]

const pad = Math.round(h * 0.01)

for (const { name, top, bottom } of regions) {
  const t = Math.max(0, Math.round(h * top) - pad)
  const b = Math.min(h, Math.round(h * bottom) + pad)
  const outPath = path.join(outputDir, name)
  await sharp(input)
    .extract({ left: 0, top: t, width: w, height: b - t })
    .toFile(outPath)
  console.log(`Saved: ${outPath}`)
}

// Square symbol with white background
const symbolT = Math.max(0, Math.round(h * 0.1) - pad)
const symbolB = Math.min(h, Math.round(h * 0.52) + pad)
const symbolH = symbolB - symbolT
const symbolW = w
const squareSize = Math.max(symbolW, symbolH)
const yOffset = Math.round((squareSize - symbolH) / 2)

const symbolSquarePath = path.join(outputDir, 'logo-symbol.png')
await sharp(input)
  .extract({ left: 0, top: symbolT, width: symbolW, height: symbolH })
  .extend({
    top: yOffset,
    bottom: squareSize - symbolH - yOffset,
    left: 0,
    right: 0,
    background: { r: 245, g: 243, b: 240, alpha: 1 },
  })
  .toFile(symbolSquarePath)
console.log(`Saved: ${symbolSquarePath}  (${squareSize}x${squareSize})`)

// Square symbol with transparent background
const symbolBuf = await sharp(input)
  .extract({ left: 0, top: symbolT, width: symbolW, height: symbolH })
  .ensureAlpha()
  .raw()
  .toBuffer({ resolveWithObject: true })

const { data, info } = symbolBuf
const pixels = Buffer.from(data)
const threshold = 240
for (let i = 0; i < pixels.length; i += 4) {
  if (pixels[i] > threshold && pixels[i + 1] > threshold && pixels[i + 2] > threshold) {
    pixels[i + 3] = 0
  }
}
const transparentPath = path.join(outputDir, 'logo-symbol-transparent.png')
await sharp(pixels, { raw: { width: info.width, height: info.height, channels: 4 } })
  .extend({
    top: yOffset,
    bottom: squareSize - symbolH - yOffset,
    left: 0,
    right: 0,
    background: { r: 0, g: 0, b: 0, alpha: 0 },
  })
  .png()
  .toFile(transparentPath)
console.log(`Saved: ${transparentPath}  (${squareSize}x${squareSize})`)

// Full copy
const fullPath = path.join(outputDir, 'logo-full.png')
await sharp(input).toFile(fullPath)
console.log(`Saved: ${fullPath}`)

console.log('\nDone!')
